FtcRobotController/src/main/java/org/firstinspires/ftc/robotcontroller/internal/SplashActivity.java

```diff
@@ -1,17 +1,175 @@
 package org.firstinspires.ftc.robotcontroller.internal;
 
 import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
 import android.content.Intent;
 import android.os.Bundle;
+import android.util.Log;
 
-public class SplashActivity extends Activity {
+import com.cyanogenmod.updater.utils.MD5;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class SplashActivity extends Activity
+{
+    File libInProtectedStorage;
+    File protectedExtraFolder;
+    File libOnSdcard;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    protected void onCreate(Bundle savedInstanceState)
+    {
         super.onCreate(savedInstanceState);
 
-        Intent intent = new Intent(this, FtcRobotControllerActivity.class);
-        startActivity(intent);
-        finish();
+        if(shouldProceedWithRcLaunch())
+        {
+            /*
+             * We've been given the go-ahead! Load up libVuforiaReal.so
+             */
+            System.load(libInProtectedStorage.getAbsolutePath());
+            Log.e("lib", "loaded!");
+
+            /*
+             * Start loading the main activity. After it's loaded, close
+             * down the splash screen
+             */
+            Intent intent = new Intent(this, FtcRobotControllerActivity.class);
+            startActivity(intent);
+            finish();
+        }
+    }
+
+    private boolean shouldProceedWithRcLaunch()
+    {
```

Comment:
I don't love all the side effects that come from this method, _especially_ when it has such an innocuous name that implies that it's just doing checks. I also don't like the true/false error state. It only works because this method is doing too much (handling errors _and_ showing UI as a result).

I'd prefer to see it set up this way: this method should be called `setupFilesForLaunch()` (or similar), and if any errors happen, throw a custom checked exception. Let onCreate decide how to handle them.